### ISSUE TITLE
Trainer launch failed on flatpak version

### ISSUE THREAD

**Issue by ahjolinna** [wyyadd/LaLa#9]
I installed the new 1.0.4 version as a flatpak (as --user) and I got this launch error, and yes I had remember to set the steam location.

![image](https://github.com/wyyadd/LaLa/assets/6993072/f7a470cb-6f6c-441c-a0d9-678f5e910329)

I did check the location and there is proton files installed...so don't know why it cant find it

![image](https://github.com/wyyadd/LaLa/assets/6993072/cc13d1a0-21ca-4b1f-8b7b-2d9a9fed8128)

**Comment by wyyadd** [comment]
Very strange.
``` dart
 Future<bool> _dirExist(String path) async {
  r eturn await Directory(path).exists();
 }

  if (protonPath == null || !await _dirExist(protonPath)) {
    throw Exception(getTranslatedText('Proton path not found.\n\nPlease ensure Proton is installed.\n\nCurrent Path: $protonPath');
  }
```
This is how i detect protonPath.
Seems your folder exist.

**Comment by ahjolinna** [comment]
could be a flatpak issue/bug, I do use a pre-release [1.15.4](https://github.com/flatpak/flatpak/releases/tag/1.15.4) version as it has some features/fixes that the stable 1.14 branch don't have. Sadly flatpak hasn't had a new release for either branches since March


PS. thanks for adding a flatpak version, I hope you will [add it to flathub](https://docs.flathub.org/docs/for-app-authors/submission/) eventually.
flatpak is kinda starting to become mandatory as Immutable Linux Distros are starting to become the new thing (SteamOS, openSUSE-[MicroOS](https://get.opensuse.org/microos/), SUSE-[ALP](https://susealp.io/), Fedora-[Silverblue](https://fedoraproject.org/silverblue/) .....[Ubuntu](https://ubuntu.com/blog/ubuntu-core-an-immutable-linux-desktop) is working on their version)

**Comment by wyyadd** [comment]
Flathub is on the way! :)
Finally I figure out why your protonPath don't exist after some tests. 

> The home permission also has exceptions as it does not grant access to the subdirectories for other applications in ~/.var/app/

LaLa cannot access .var/app folder according to [Flatpak filesystem-access](https://docs.flatpak.org/en/latest/sandbox-permissions.html#filesystem-access)

**Comment by wyyadd** [comment]
Update: 
After install steam through [multilib](https://wiki.archlinux.org/title/Multilib) repository, I successfully launched trainer.
![Screenshot from 2023-11-06 21-20-20](https://github.com/wyyadd/LaLa/assets/67038748/9ebd6ed7-4018-4ed5-99b2-adcbaaaec8c0)

**Comment by wyyadd** [comment]
Thanks for the information.
After adding --filesystem=~/.var/app/com.valvesoftware.Steam/ to manifest, LaLa can access steam folder and launch trainers successfully.

**Comment by wyyadd** [comment]
I fixed this issue In release [v1.0.5](https://github.com/wyyadd/LaLa/releases/tag/v1.0.5).
So, mark this as closed.
